tests/integration/search/test_search_views.py

```diff
@@ -466,3 +466,38 @@ def test_search_hep_author_publication_with_not_existing_facet_name(isolated_api
         assert aggregation in result_aggregations
 
     assert len(expected_aggregations) == len(result_aggregations)
+
+
+def test_selecting_2_facets_generates_search_with_must_query(isolated_api_client):
+    record_json = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        'document_type': ['article'],
+        'titles': [{'title': 'Article 1'}],
+        '_collections': ['Literature'],
+        'authors': [{'full_name': 'John Doe'}]
+    }
+
+    rec = InspireRecord.create(data=record_json)
+    rec.commit()
+
+    record_json2 = {
+        '$schema': 'http://localhost:5000/schemas/records/hep.json',
+        'document_type': ['article'],
+        'titles': [{'title': 'Article 2'}],
+        '_collections': ['Literature'],
+        'authors': [{'full_name': 'John Doe'}, {'full_name': 'John Doe2'}]
+    }
+
+    rec2 = InspireRecord.create(data=record_json2)
+    rec2.commit()
+
+    db.session.commit()
+    es.indices.refresh('records-hep')
+
+    response = isolated_api_client.get('/literature?q=&author=John%20Doe')
+    data = json.loads(response.data)
+    assert data["hits"]["total"] == 2
```

Comment:
I'm this will fail because we have no isolation, instead can we check that both records are in the results instead of checking the number?